File: python/3477.fruits-into-baskets-ii.py
class Solution:
    def numOfUnplacedFruits(self, fruits: list[int], baskets: list[int]) -> int:
        left = len(baskets)

        for fruit in fruits:
            for j in range(len(baskets)):
                if fruit <= baskets[j]:
                    baskets[j] = -1
                    left -= 1
                    break

        return left

File: cpp/3477.fruits-into-baskets-ii.cpp
#include <vector>
using std::vector;

class Solution {
public:
  int numOfUnplacedFruits(vector<int> &fruits, vector<int> &baskets) {
    const int n = baskets.size();
    int left = n;

    for (int fruit : fruits) {
      for (int j = 0; j < n; j++) {
        if (fruit <= baskets[j]) {
          baskets[j] = -1;
          left--;
          break;
        }
      }
    }

    return left;
  }
};